File: chroma-league/src/test/java/com/bonepl/chromaleague/tasks/FetchGameStatsTaskTest.java
package com.bonepl.chromaleague.tasks;

import com.bonepl.chromaleague.rest.http.LeagueHttpClientMock;
import com.bonepl.chromaleague.state.RunningState;
import org.junit.jupiter.api.Test;

import static com.bonepl.chromaleague.rest.http.handlers.GameStatsResponseHandlerTest.verifyGameStats;

class FetchGameStatsTaskTest {

    @Test
    void testGameStatsParsing() {
        //given
        new LeagueHttpClientMock().mockGameStatsResponse("json/gamestats.json");

        //when
        new FetchGameStatsTask().run();

        //then
        verifyGameStats(RunningState.getGameState().getGameStats());
    }
}

File: test/tasks/FetchGameStatsTask.test.ts
import {beforeEach, describe, expect, test, vi} from 'vitest';
import {RunningState} from '../../src/state/RunningState.js';
import type {GameStats} from '../../src/rest/gamestats/GameStats.js';

const gameStatsJson: GameStats = {
    gameMode: 'CLASSIC',
    gameTime: 15.123456789,
    mapTerrain: 'Infernal',
};

vi.mock('../../src/rest/http/LeagueHttpClients.js', () => ({
    getNonBlockingResponse: vi.fn().mockResolvedValue(gameStatsJson),
    getRetryingResponse: vi.fn().mockResolvedValue(gameStatsJson),
    setNonBlockingLeagueHttpClient: vi.fn(),
    setRetryingLeagueHttpClient: vi.fn(),
}));

describe('FetchGameStatsTask', () => {
    beforeEach(() => {
        RunningState.setRunningGame(false);
        RunningState.setRunningGame(true);
    });

    test('testGameStatsParsing', async () => {
        //given & when
        const { fetchGameStatsTask } = await import('../../src/tasks/FetchGameStatsTask.js');
        await fetchGameStatsTask();

        //then
        const gameStats = RunningState.getGameState().gameStats!;
        expect(gameStats).toBeDefined();
        expect(gameStats.gameMode).toBe('CLASSIC');
        expect(gameStats.gameTime).toBe(15.123456789);
        expect(gameStats.mapTerrain).toBe('Infernal');
    });
});
